Enable rtc on rt700

Holt-Sun opened the pull request:
Updated rtc driver to support NXP RT700.
1. Update nxp irtc driver to fix issue in init and alarm get function.
2. Update RTC device tree binding to support "share-counter".
3. Update RT700 dtsi to support rtc0 for cpu0 and rtc1 for cpu1.
4. Update readme.

Enable rt700 RTC unittest.

bjarki-andreasen commented on samples/drivers/rtc/src/main.c:
```diff
@@ -11,6 +11,7 @@
 
 const struct device *const rtc = DEVICE_DT_GET(DT_ALIAS(rtc));
 
+#if !(DT_PROP(DT_ALIAS(rtc), share_counter))
```
this can not go in a common sample, board/soc specific logic stays with the board/soc, if really needed, in a board specific sample in `samples/boards/nxp/` :)

Holt-Sun replied:
Updated. Move the `#if (DT_PROP(DT_ALIAS(rtc), share_counter))` into rtc_nxp_irtc.c `nxp_irtc_set_time` function. Please review.

bjarki-andreasen commented on dts/bindings/rtc/nxp,irtc.yaml:
```diff
@@ -40,3 +39,9 @@ properties:
       Default value ensures the output clock is turned off.
       This is the reset value.
+  
+  share-counter:
+    type: boolean
+    description: |
+      This slave irtc instance shares the data and time counters of the master irtc instance. 
+      This means the code cannot set the data and time counters, but can only read them.
```
This property should be enforced on the driver level, by implementing the set_time API like
```
static int set_time_not_perm(const struct device *dev, const struct rtc_time *timeptr)
{
        ARG_UNUSED(timeptr);
        return -EPERM;
}

static DEVICE_API(...) = {
        .set_time = set_time_not_sup,
        ...
};
```
the sample does not need to be updated, it will correclty print "Cannot write date time: -1"

bjarki-andreasen commented on dts/bindings/rtc/nxp,irtc.yaml:
```diff
@@ -40,3 +39,9 @@ properties:
+  share-counter:
+    type: boolean
+    description: |
+      This slave irtc instance shares the data and time counters of the master irtc instance. 
```
use a different term than "slave" and "master". See https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#rule-a-2-inclusive-language

zejiang0jason commented on drivers/rtc/rtc_nxp_irtc.c:
```diff
@@ -79,12 +86,12 @@ static int nxp_irtc_set_time(const struct device *dev, const struct rtc_time *ti
-	irtc_reg->HOURMIN = RTC_HOURMIN_MIN_CNT(timeptr->tm_min) |
-		RTC_HOURMIN_HOUR_CNT(timeptr->tm_hour);
+	irtc_reg->HOURMIN =
+		RTC_HOURMIN_MIN_CNT(timeptr->tm_min) | RTC_HOURMIN_HOUR_CNT(timeptr->tm_hour);
```
Just confirm, is this result by the format script? thanks.

Holt-Sun replied:
Yes, it is format by clang-format with root .

zejiang0jason commented on drivers/rtc/rtc_nxp_irtc.c:
```diff
@@ -375,17 +412,40 @@ static DEVICE_API(rtc, rtc_nxp_irtc_driver_api) = {
-		.clock_src = DT_INST_PROP(n, clock_src),					   \
-		.is_output_clock_enabled = DT_INST_PROP(n, output_clk_en),	                   \
+		NXP_IRTC_CLOCK_SELECTION_INIT(n).is_output_clock_enabled =                         \
+			DT_INST_PROP(n, output_clk_en),                                            \
```
Should this be split to two lines? like:
```
		NXP_IRTC_CLOCK_SELECTION_INIT(n) \
                .is_output_clock_enabled = DT_INST_PROP(n, output_clk_en),
```

Holt-Sun replied:
This is the formated code.
I have split to 2 lines.

Holt-Sun committed:
drivers: irtc: Updated rtc driver to support NXP RT700 device.

1. Update nxp irtc driver to fix issue in init and alarm function.
2. Update RTC device tree binding to support "share-counter".
3. Update RT700 dtsi to support rtc0 for cpu0 and rtc1 for cpu1.
4. Update readme.
5. Update unit test project conf for RT700.

Signed-off-by: Holt Sun <holt.sun@nxp.com>

bjarki-andreasen reviewed (commented):
Changes look fine, next are the CI failures :)

Holt-Sun commented on drivers/rtc/rtc_nxp_irtc.c:
```diff
@@ -62,6 +63,12 @@ static void nxp_irtc_unlock_registers(RTC_Type *reg)
+	ARG_UNUSED(timeptr);
+
+	return -EPERM;
+#else
```
Updated code to add a bool share_counter.

Holt-Sun commented:
> Changes look fine, next are the CI failures :)
Hi @bjarki-andreasen , from https://github.com/zephyrproject-rtos/zephyr/pull/93006/checks, there are no CI failures. Thanks.
